File: core/src/main/scala/com/graphbrain/db/TextNode.scala
package com.graphbrain.db

case class TextNode(override val id: String,
                    override val summary: String="",
                    override val degree: Int = 0,
                    override val ts: Long = -1)
  extends Textual(id, summary, degree, ts) {

  def this(id: String, map: Map[String, String]) =
    this(id,
      map("summary"),
      map("degree").toInt,
      map("ts").toLong)

  override def extraMap = Map("summary" -> summary)

  override def setId(newId: String): Vertex = copy(id=newId)

  override def setDegree(newDegree: Int): Vertex = copy(degree=newDegree)

  override def setTs(newTs: Long): Vertex = copy(ts=newTs)

  /*
  override def updateSummary: Textual = TextNode(namespace, text, generateSummary)


  override def updateFromEdges(): Vertex = {
    val newSummary = generateSummary
    val newVertex = this.copy(summary=newSummary)

    if (this != newVertex) {
      put()
    }

    this
  }*/

  override def raw: String = {
    "type: " + "text<br />" +
    "id: " + id + "<br />" +
    "summary: " + summary + "<br />"
  }
}


object TextNode {
  def fromNsAndText(namespace: String,
    text: String,
    summary: String="",
    degree: Int = 0,
    ts: Long = -1) = {

    TextNode(namespace + "/" + ID.sanitize(text).toLowerCase,
      summary,
      degree,
      ts)
  }
}

File: core/src/main/java/com/graphbrain/db/TextNode.java
package com.graphbrain.db;

public class TextNode extends Vertex {

    private String text;

    @Override
    public VertexType type() {return VertexType.Text;}

    public TextNode(String id, String text, int degree, long ts) {
        super(id, degree, ts);
        this.text = text;
    }

    public TextNode(String text, int degree, long ts) {
        this(idFromText(text), text, degree, ts);
    }

    public TextNode(String id, String text) {
        this(id, text, 0, -1);
    }

    @Override
    public Vertex copy() {
        return new TextNode(id, text, degree, ts);
    }

    public String getText() {
        return text;
    }

    public static String idFromText(String text) {
        String shortText = text.substring(0, Math.min(text.length(), 50));
        String hash = ID.hash(text);
        return hash + "/" + ID.sanitize(shortText);
    }
}